File: src/main/java/com/oracle/ocmjea/gof/structural/adapter/MediaAdapter.java
package com.oracle.ocmjea.gof.structural.adapter;

public class MediaAdapter implements MediaPlayer {

	private MediaPlayerAvancado playerAvancado;

	public MediaAdapter(String tipoAudio) {
		switch (tipoAudio) {
		case "flv":
			playerAvancado = new FlvPlayer();
		case "mp4":
			playerAvancado = new Mp4Player();
		}
	}

	@Override
	public void play(String tipo, String arquivo) {
		if (tipo.equalsIgnoreCase("flv")) {
			playerAvancado.playFlv(arquivo);
		} else if (tipo.equals("mp4")) {
			playerAvancado.playMp4(arquivo);
		}
	}
}

File: design-patterns/src/main/kotlin/designpatterns/structural/adapter/MediaAdapter.kt
package designpatterns.structural.adapter

class MediaAdapter(audioType: String) : MediaPlayer {
    private val advancedPlayer: AdvancedMediaPlayer? = when (audioType) {
        "flv" -> FlvPlayer()
        "mp4" -> Mp4Player()
        else -> null
    }

    override fun play(type: String, fileName: String) {
        if (type.equals("flv", ignoreCase = true)) {
            advancedPlayer?.playFlv(fileName)
        } else if (type == "mp4") {
            advancedPlayer?.playMp4(fileName)
        }
    }
}
